--- gdata-2.0.17/src/gdata/youtube/data.py.orig	2013-01-10 23:44:27.751179610 +0100
+++ gdata-2.0.17/src/gdata/youtube/data.py	2013-01-10 23:44:35.047910614 +0100
@@ -30,7 +30,7 @@
 import gdata.youtube.data
 
 
-YT_TEMPLATE = '{http://gdata.youtube.com/schemas/2007/}%s'
+YT_TEMPLATE = '{http://gdata.youtube.com/schemas/2007}%s'
 
 
 class ComplaintEntry(gdata.data.GDEntry):
--- gdata-2.0.17/src/gdata/media/data.py.orig	2013-01-10 23:44:15.480631603 +0100
+++ gdata-2.0.17/src/gdata/media/data.py	2013-01-10 23:44:21.127423660 +0100
@@ -23,7 +23,7 @@
 import atom.core
 
 
-MEDIA_TEMPLATE = '{http://search.yahoo.com/mrss//}%s'
+MEDIA_TEMPLATE = '{http://search.yahoo.com/mrss/}%s'
 
 
 class MediaCategory(atom.core.XmlElement):
